src/libraries/System.Private.CoreLib/src/System/IO/BinaryWriter.cs

```diff
@@ -453,17 +453,52 @@ public virtual void Write(ReadOnlySpan<char> chars)
             }
         }
 
-        protected void Write7BitEncodedInt(int value)
+        public void Write7BitEncodedInt(int value) => Write7BitEncodedUInt32((uint)value);
+
+        public void Write7BitEncodedInt64(long value) => Write7BitEncodedUInt64((ulong)value);
+
+        private void Write7BitEncodedUInt32(uint value)
+        {
+            // On 64-bit platforms, always use the 64-bit overload.
+            // It's just a small optimization that allows AOT compilers
+            // to eliminate this method implementation entirely.
```

Comment:
I would expect this to be a deoptimization in the typical case. `Write7BitEncodedUInt64` is going to be used very rarely. This will keep it around in the typical case.

Reply:
What I was trying to go for is that on 64-bit platforms only the method `Write7BitEncodedUInt64` sticks around - everything else has been pruned away or inlined. If this can't be guaranteed then yeah, this is probably just unnecessary complexity. :(